### ISSUE TITLE
SelectBooleanCheckbox: Rendering bug in EdgeC

### ISSUE THREAD

**Issue by fanste** [primefaces/primefaces#7858]
**Describe the defect**
I have a weird bug in my layout which I'm currently not able to transfer to a small reproducer.
_Edit: Reproducable in showcase - see https://github.com/primefaces/primefaces/issues/7858#issuecomment-925940074_

My layout is based on CSS flexbox (100% height) with a scrollable content area. The body has `overflow:hidden`. As soon as a `p:selectBooleanCheckbox` is used the hidden `input[type=checkbox]` is rendered at the original position n the client window because of its `position:absolute` in `(.ui-helper-hidden-accessible)` regardless of the scrollable area.

If the checkbox would be at y=2000px while the scrollable area only has h=500px, the hidden input is always outside the visible space. Its absolute position won't be affected by the scrollbar.

When you click the checkbox/label, the browser/widget tries to`focus` on that hidden input and the browser goes nuts! Why? Because the focused hidden input is somewhere in the client window but not in the visible space.
![image](https://user-images.githubusercontent.com/3531183/134534857-98bee787-779f-4fc4-a46c-29d5227a0259.png)

The space that is generated in `body` by the hidden input is even visible if `overflow:hidden` is removed on `body`.
![image](https://user-images.githubusercontent.com/3531183/134535405-b86e27a0-e12d-423b-aa11-d919809b0034.png)

The second scrollbar disappears as soon as the checkbox is removed.

**Environment:**
 - PF Version: _10.0_
 - Affected browsers: MS Edge (Chrome)

**Possible fix**
Add `position:relative` to `.ui-selectbooleancheckbox`. Or maybe to `.ui-chkbox`, but I don't know if this will have negative effects on other checkbox-like components or renderings.

**Comment by melloware** [comment]
You didn't mention what Theme you are using.  You might not be able to reproduce it with PrimeFaces test because its using Saga theme?

**Comment by fanste** [comment]
We use an older custom theme based on Themeroller, that was adjusted from time to time to support the new components - yeah I know, it should be totally reworked ;)

But the theme in the showcase has the same problem - maybe I missed something in my reproducer.


Execute the following javascript (DevTools) in the [showcase ](https://www.primefaces.org/showcase/ui/input/booleanCheckbox.xhtml)and click the `Basic` checkbox, you will see the same behaviour:
```Javascript
$('h1').css('margin-bottom', '500px')
$('.layout-content').css({'overflow-y': 'auto', 'height':'500px'})
$('body').css('overflow-y', 'hidden')
```
Without the last command you will even see the second scrollbar and how it scrolls to the hidden input after you click the checkbox.

This command fixes the `Basic` checkbox, while the `Ajax` checkbox is still broken.
```Javascript
$('.ui-selectbooleancheckbox:first').css('position', 'relative')
```

**Comment by melloware** [comment]
OK this is @mertsincan territory with why the CSS is like that.  I always thought `ui-helper-hidden-accessible` hid it from everything.

**Comment by melloware** [comment]
@fanste does this still happen? I just tried in the showcase page using the exact steps you did in Chrome and Edge and can't get it to happen?  Just making sure I am not doing something wrong or its possible a Chrome bug was resolved?

**Comment by fanste** [comment]
https://github.com/primefaces/primefaces/blame/16fad745ba4f5a222a05af9105db5804d6035df1/primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.css#L118

Fixed with #7884 => PF version 11

**Comment by melloware** [comment]
Nice find!
